### PR TITLE
feature/decouple adapters from core

### PR DESCRIPTION
**Pull request by colin-rogers-dbt**
resolves #
[docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose) dbt-labs/docs.getdbt.com/#

### Checklist

- [ ] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [ ] I have run this code in development and it appears to resolve the stated issue
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [ ] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX

### CODE REVIEW

File: dbt/adapters/bigquery/connections.py [hunk 1]
```diff
@@ -382,7 +382,7 @@ def get_bigquery_client(cls, profile_credentials):
         execution_project = profile_credentials.execution_project
         location = getattr(profile_credentials, "location", None)
 
-        info = client_info.ClientInfo(user_agent=f"dbt-{dbt_version}")
```
Comment (colin-rogers-dbt) [inline dbt/adapters/bigquery/connections.py:385]: We can no longer import the dbt-core version so will supply the dbt-bigquery one

File: dbt/adapters/bigquery/relation_configs/_base.py [hunk 2]
```diff
@@ -1,17 +1,17 @@
 from dataclasses import dataclass
-from typing import Optional
+from typing import Optional, Dict
 
 import agate
 from dbt.adapters.base.relation import Policy
 from dbt.adapters.relation_configs import RelationConfigBase
 from google.cloud.bigquery import Table as BigQueryTable
+from typing_extensions import Self
```
[the remark this answers is not in the document]
Comment (colin-rogers-dbt) [reply dbt/adapters/bigquery/relation_configs/_base.py:8]: I believe `Self` wasn't introduced until 3.11 so we need to use `typing_extensions` for compatibility w earlier versions

File: dbt/adapters/bigquery/impl.py [hunk 3]
```diff
@@ -145,7 +147,9 @@ def drop_relation(self, relation: BigQueryRelation) -> None:
         conn.handle.delete_table(table_ref, not_found_ok=True)
 
     def truncate_relation(self, relation: BigQueryRelation) -> None:
-        raise dbt.exceptions.NotImplementedError("`truncate` is not implemented for this adapter!")
+        raise dbt.common.exceptions.base.NotImplementedError(
```
[the remark this answers is not in the document]
Comment (colin-rogers-dbt) [reply dbt/adapters/bigquery/impl.py:150]: this is tied into our event logging system so we want our own customer exception here

File: dbt/adapters/bigquery/impl.py [hunk 4]
```diff
@@ -160,7 +164,7 @@ def rename_relation(
             or from_relation.type == RelationType.View
             or to_relation.type == RelationType.View
         ):
-            raise dbt.exceptions.DbtRuntimeError(
+            raise dbt.common.exceptions.DbtRuntimeError(
```
[the remark this answers is not in the document]
Comment (colin-rogers-dbt) [reply dbt/adapters/bigquery/impl.py:167]: Agreed, we should add a new adapter exception, something like `UnsupportedOperationForPlatform` but that's out of scope for this change

File: dbt/adapters/bigquery/impl.py [hunk 5]
```diff
@@ -679,14 +682,16 @@ def upload_file(
             self.poll_until_job_completes(job, timeout)
 
     @classmethod
-    def _catalog_filter_table(cls, table: agate.Table, manifest: Manifest) -> agate.Table:
+    def _catalog_filter_table(
+        cls, table: agate.Table, used_schemas: FrozenSet[Tuple[str, str]]
```
[the remark this answers is not in the document]
Comment (colin-rogers-dbt) [reply dbt/adapters/bigquery/impl.py:686]: that's what we've been using elsewhere. I don't _think_ it does tho

**Review by mikealfare — APPROVED** [review]

**Pull request merged**